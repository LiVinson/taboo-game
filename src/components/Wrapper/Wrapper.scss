.wrapper {
    background-image: linear-gradient(to bottom, var(--color-white) 0%, var(--color-white) 50%,
         70%, var(--color-primary) 90%);
    padding: 1rem 0rem 6rem 0rem;
    min-height:100vh;
    position: relative;
    overflow:hidden; 

}