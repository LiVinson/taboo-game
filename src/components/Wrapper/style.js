import styled from "styled-components"

export const StyledWrapper = styled.div`
  padding: 1rem 0rem 6rem 0rem;
  min-height: 100vh;
  position: relative;
  /* Needs z-index so contents inside will be aboe layered taboo cards */
  z-index:0;
  width: 30rem;
  max-width:100%;
  margin: 0 auto;  
`
