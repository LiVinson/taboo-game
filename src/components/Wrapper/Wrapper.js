import React from "react"
import PropTypes from "prop-types"
import { StyledWrapper } from "./style"

const Wrapper = ({ children }) => {
    return (
        <StyledWrapper>
            {children}
        </StyledWrapper>
    )
}

Wrapper.propTypes = {
    children: PropTypes.node.isRequired
}

export default Wrapper