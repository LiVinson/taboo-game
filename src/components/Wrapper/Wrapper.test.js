import React from "react"
import { shallow } from "enzyme"
import renderer from "react-test-renderer"
import Wrapper from "./Wrapper"

it("renders correctly", ()=> {
    const children = <h1>I am a child element</h1>
    const WrapperComponent = renderer.create(<Wrapper>{children}</Wrapper>).toJSON()
    expect(WrapperComponent).toMatchSnapshot()
})

it("renders props.children", ()=> {
    const children = <h1>I am a child element</h1>
    const WrapperComponent = shallow(<Wrapper>{children}</Wrapper>)
    expect(WrapperComponent.contains(children)).toBe(true)
})