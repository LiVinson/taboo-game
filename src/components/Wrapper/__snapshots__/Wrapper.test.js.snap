// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper"
>
  <h1>
    I am a child element
  </h1>
</div>
`;
